WordPress/openverse-api
# Upgrade Django from 4.1.0 to 4.1.2 for security fix

## Description


This PR addresses a security issue detected by dependabot which dependabot was not able to resolve on its own. This is because the `django-oauth-toolkit` library also needed to be updated alongside Django due to dependency constraints the former had specified.

**Related Changelogs**:
- Django: https://docs.djangoproject.com/en/4.1/releases/4.1.2/
- Django OAuth Toolkit: https://github.com/jazzband/django-oauth-toolkit/blob/master/CHANGELOG.md#220-2022-10-18

No migrations are needed for this upgrade.

## Testing Instructions

1. `just api-test`
1. Run the API locally and ensure that normal actions can be completed.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
